src/Nethermind/Nethermind.Trie/ITreeVisitor.cs

```diff
@@ -29,4 +30,75 @@ public interface ITreeVisitor
 
         void VisitCode(Hash256 codeHash, TrieVisitContext trieVisitContext);
     }
+
+    public interface ITreeVisitor<TNodeContext>
+        where TNodeContext : struct, INodeContext<TNodeContext>
+    {
+        /// <summary>
+        /// Specify that this is a full table scan and should optimize for it.
+        /// </summary>
+        public bool IsFullDbScan { get; }
```

Comment:
Fixed per type? If so use static interface method? (not sure if exactly this)
```csharp
static abstract IsFullDbScan { get; } 
```

Reply:
I'd leave to a visitor. Some can have the instance dependent logic.